conda_lock/src_parser/__init__.py

```diff
@@ -169,7 +169,10 @@ def create(
         metadata_choices: AbstractSet[MetadataOption],
         src_files: List[pathlib.Path],
     ) -> "GitMeta | None":
-        import git
+        try:
+            import git
+        except ImportError:
+            return None
```

Comment:
```suggestion
            raise RuntimeError("The GitPython package is required to read Git metadata.")
```

Reply:
Wait, that's probably not right, as `GitMeta.create` gets [created unconditionally](https://github.com/conda-incubator/conda-lock/blob/90b370a7b8f2b918820159a3c4f95ac0a31ac446/conda_lock/conda_lock.py#LL826C5-L826C17).

Reply:
Right.. it's not computed on-demand like I originally thought, so this is going to require more changes. :slightly_frowning_face:

Reply:
rolled back... would a `warning` be more appropriate?

Reply:
I feel like this should be a fatal error; the metadata is explicitly requested, so I think it's much better to hard fail. Here's another attempt:
https://github.com/bollwyvl/conda-lock/pull/1

I still need to run pytest on it though.

Reply:
Running tests via CI here:
https://github.com/conda-incubator/conda-lock/pull/298

Reply:
It's already marked optional all over: as in, `conda-lock` doesn't fail in a non-git checkout.

Reply:
Right, but if a user runs
```
conda-lock --metadata git_user_name
```
then it would be mean to the user to silent fail, right? Or am I missing something obvious?